esphome/components/duty_cycle/output.py

```diff
@@ -0,0 +1,29 @@
+from esphome import pins
+from esphome.components import output
+import esphome.config_validation as cv
+import esphome.codegen as cg
+from esphome.const import CONF_ID, CONF_PIN, CONF_PERIOD
+
+duty_cycle_ns = cg.esphome_ns.namespace("duty_cycle")
+DutyCycleOutput = duty_cycle_ns.class_(
+    "DutyCycleOutput", output.FloatOutput, cg.Component
+)
+validate_frequency = cv.All(cv.frequency, cv.Range(min=1.0e-6))
+
+CONFIG_SCHEMA = output.FLOAT_OUTPUT_SCHEMA.extend(
+    {
+        cv.Required(CONF_ID): cv.declare_id(DutyCycleOutput),
+        cv.Required(CONF_PIN): pins.gpio_output_pin_schema,
+        cv.Required(CONF_PERIOD): cv.positive_time_period_milliseconds,
```

Comment:
```suggestion
        cv.Required(CONF_PERIOD): cv.Range(cv.positive_time_period_milliseconds, core.TimePeriod(milliseconds='100ms')),
```

It would be good to validate the time period is not too low - at a certain point the component will not work accurately (and I fear some will try to use this component mistakenly for fast-switchting outputs).

The main loop runs at 60Hz - so 100ms is a good lower bound.

Reply:
Sounds good to me. 

As a side note: Since I will rename this component to `slow_pwm` do you think it makes sense to still use `period` as the config here? Do you think it should be `frequency` to match `pwm`? 

My opinion is that `period` is a more ergonomic option here since typical use would be something like `1Hz`, `0.5Hz`, etc, if it were to be `frequency`, but just want to see what your opinion was.

Reply:
Yeah I think period is much better here - otherwise when you want a period of say 1min you'd need to type `.0166Hz` which is not that nice.

Plus it even incentivizes the use of long periods - whereas frequency would do the reverse.